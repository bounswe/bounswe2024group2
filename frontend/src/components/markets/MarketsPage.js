import React from 'react';
import { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import '../../styles/MarketsPage.css';
import { mockIndices } from './MockData';

const MarketsPage = () => {
    const [selectedIndex, setSelectedIndex] = useState(null);
    
    const navigate = useNavigate();

    const handleIndexClick = (id) => {
        navigate(`/stocks/${id}`);
    };
    return (
        <div className="markets-container">
      <h1>Indices</h1>
      <div className="indices">
        {mockIndices.map(index => (
          <div 
            key={index.id} 
            onClick={() => handleIndexClick(index.id)} 
            className="index-item"
          >
            {index.name}
          </div>
        ))}
      </div>
    </div>
  );
}

export default MarketsPage;

