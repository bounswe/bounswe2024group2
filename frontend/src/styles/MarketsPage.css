.markets-container {
  padding: 20px;
  position: absolute;
  top: 30px;
  left: 0;
}


.indices {
  display: flex;
  gap: 20px; 
  flex-wrap: wrap; 
}

.index-item {
  cursor: pointer;
  padding: 20px;
  border: 2px solid black;
  width: 150px;
  text-align: center;
  border-radius: 8px;
  background-color: #f5f5f5;
  transition: background-color 0.3s ease, box-shadow 0.3s ease; 
}

.index-item:hover {
  background-color: #007bff;
  color: white; 
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.2);
}
