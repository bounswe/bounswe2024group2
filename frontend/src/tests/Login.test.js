// MyComponent.test.js
import React from 'react';
import { render } from '@testing-library/react';
import MarketsPage from '../components/markets/MarketsPage';

test('renders Login correctly', () => {
  const { asFragment } = render(<MarketsPage title="Snapshot Testing" />);
  
  expect(asFragment()).toMatchSnapshot();
});
