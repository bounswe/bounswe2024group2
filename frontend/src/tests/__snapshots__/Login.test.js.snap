// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Login correctly 1`] = `
<DocumentFragment>
  <div
    class="news-page"
  >
    <div
      class="news-header"
    >
      <h1
        class="news-title"
      >
        Economic News
      </h1>
      <h2
        class="news-subtitle"
      >
        Your daily updates on economic trends
      </h2>
    </div>
    <div
      class="news-content"
    >
      <h3
        class="section-title"
      >
        Mock Economic News Section
      </h3>
      <div
        class="filter-buttons"
      >
        <div
          class="filter-buttons"
        >
          <button
            class="filter-button selected"
          >
            All
          </button>
          <button
            class="filter-button "
          >
            Global Economy
          </button>
          <button
            class="filter-button "
          >
            Stock Market
          </button>
          <button
            class="filter-button "
          >
            Cryptocurrency
          </button>
          <button
            class="filter-button "
          >
            Inflation
          </button>
          <button
            class="filter-button "
          >
            Job Market
          </button>
        </div>
      </div>
      <div
        class="scrollable-section"
      >
        <div
          class="card"
        >
          <div
            class="card-header"
          >
            <span
              class="card-source"
            >
              Economic Times
            </span>
            <span
              class="card-time"
            >
              2 hours ago
            </span>
          </div>
          <div
            class="card-body"
          >
            <div
              class="card-text"
            >
              <h3
                class="card-title"
              >
                GDP Growth Forecasts Revised
              </h3>
              <p
                class="card-description"
              >
                Recent reports indicate that GDP growth forecasts for major economies have been revised upwards.
              </p>
            </div>
            <div
              class="card-icon"
            >
              <img
                alt="Economic Times"
                src="https://www.thisiscolossal.com/wp-content/uploads/2024/10/jonk-1.jpg"
              />
            </div>
          </div>
        </div>
        <div
          class="card"
        >
          <div
            class="card-header"
          >
            <span
              class="card-source"
            >
              Market Watch
            </span>
            <span
              class="card-time"
            >
              3 hours ago
            </span>
          </div>
          <div
            class="card-body"
          >
            <div
              class="card-text"
            >
              <h3
                class="card-title"
              >
                Major Indices Reach All-Time Highs
              </h3>
              <p
                class="card-description"
              >
                The stock market sees a significant uptick as major indices reach all-time highs.
              </p>
            </div>
          </div>
        </div>
        <div
          class="card"
        >
          <div
            class="card-header"
          >
            <span
              class="card-source"
            >
              CoinDesk
            </span>
            <span
              class="card-time"
            >
              1 hour ago
            </span>
          </div>
          <div
            class="card-body"
          >
            <div
              class="card-text"
            >
              <h3
                class="card-title"
              >
                Bitcoin Surges Past $60,000
              </h3>
              <p
                class="card-description"
              >
                Bitcoin's value has surged past $60,000 amid increased institutional adoption.
              </p>
            </div>
          </div>
        </div>
        <div
          class="card"
        >
          <div
            class="card-header"
          >
            <span
              class="card-source"
            >
              Bloomberg
            </span>
            <span
              class="card-time"
            >
              4 hours ago
            </span>
          </div>
          <div
            class="card-body"
          >
            <div
              class="card-text"
            >
              <h3
                class="card-title"
              >
                Rising Costs Impact Consumers
              </h3>
              <p
                class="card-description"
              >
                Inflation rates are rising, leading to increased costs for consumers on everyday goods.
              </p>
            </div>
          </div>
        </div>
        <div
          class="card"
        >
          <div
            class="card-header"
          >
            <span
              class="card-source"
            >
              The Wall Street Journal
            </span>
            <span
              class="card-time"
            >
              5 hours ago
            </span>
          </div>
          <div
            class="card-body"
          >
            <div
              class="card-text"
            >
              <h3
                class="card-title"
              >
                Unemployment Rates Hit Record Low
              </h3>
              <p
                class="card-description"
              >
                The job market strengthens as unemployment rates hit a record low across the nation.
              </p>
            </div>
            <div
              class="card-icon"
            >
              <img
                alt="The Wall Street Journal"
                src="https://1finance.co.in/magazine/wp-content/uploads/2024/01/Blog-5-b-scaled.jpg"
              />
            </div>
          </div>
        </div>
      </div>
      <h3
        class="section-title"
      >
        Mock RSS Feed Economic News Section
      </h3>
      <div
        class="filter-buttons"
      >
        <div
          class="filter-buttons"
        >
          <button
            class="filter-button selected"
          >
            All
          </button>
          <button
            class="filter-button "
          >
            Trade
          </button>
          <button
            class="filter-button "
          >
            Market Trends
          </button>
          <button
            class="filter-button "
          >
            Federal Reserve
          </button>
        </div>
      </div>
      <div
        class="scrollable-section"
      >
        <div
          class="card"
        >
          <div
            class="card-header"
          >
            <span
              class="card-source"
            >
              Reuters
            </span>
            <span
              class="card-time"
            >
              3 days ago
            </span>
          </div>
          <div
            class="card-body"
          >
            <div
              class="card-text"
            >
              <h3
                class="card-title"
              >
                New Agreements Boost International Trade
              </h3>
              <p
                class="card-description"
              >
                New trade agreements are set to boost international trade among participating countries.
              </p>
            </div>
          </div>
        </div>
        <div
          class="card"
        >
          <div
            class="card-header"
          >
            <span
              class="card-source"
            >
              Forbes
            </span>
            <span
              class="card-time"
            >
              2 days ago
            </span>
          </div>
          <div
            class="card-body"
          >
            <div
              class="card-text"
            >
              <h3
                class="card-title"
              >
                Emerging Markets on the Rise
              </h3>
              <p
                class="card-description"
              >
                Emerging markets are gaining traction as investors look for new opportunities.
              </p>
            </div>
          </div>
        </div>
        <div
          class="card"
        >
          <div
            class="card-header"
          >
            <span
              class="card-source"
            >
              CNBC
            </span>
            <span
              class="card-time"
            >
              1 day ago
            </span>
          </div>
          <div
            class="card-body"
          >
            <div
              class="card-text"
            >
              <h3
                class="card-title"
              >
                Interest Rates Held Steady
              </h3>
              <p
                class="card-description"
              >
                The Federal Reserve has announced that interest rates will remain steady for the foreseeable future.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
